specification/cdn/resource-manager/Microsoft.Cdn/stable/2019-04-15/cdn.json

```diff
@@ -3230,6 +3231,109 @@
         }
       }
     },
+    "DeliveryRuleUrlSigningAction": {
+      "description": "Defines the Url Signing action for the delivery rule.",
+      "x-ms-discriminator-value": "UrlSigning",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DeliveryRuleAction"
+        }
+      ],
+      "required": [
+        "parameters"
+      ],
+      "properties": {
+        "parameters": {
+          "description": "Defines the parameters for the action.",
+          "$ref": "#/definitions/UrlSigningParameters"
+        }
+      }
+    },
+    "UrlSigningParameters": {
+      "description": "Defines the parameters for the Url Signing action.",
+      "required": [
+        "keys",
+        "hmacAlgorithm",
+        "queryStringParams",
+        "useIpInHash"
```

Comment:
>useIpInHash [](start = 9, length = 11)

still undecided if we should this or CIDR ranges.

Reply:
it will be CIDR range. resolved in new PR>

---
In reply to: [302758045](https://github.com/Azure/azure-rest-api-specs/pull/6438#discussion_r302758045) [](ancestors = 302758045)